e2e/validators.go

```diff
@@ -451,7 +451,85 @@ func ValidateContainerd2Properties(ctx context.Context, s *Scenario, versions []
 func ValidateContainerRuntimePlugins(ctx context.Context, s *Scenario) {
 	// nri plugin is enabled by default
 	ValidateDirectoryContent(ctx, s, "/var/run/nri", []string{"nri.sock"})
+}
 
+func ValidateNPDGPUCountPlugin(ctx context.Context, s *Scenario) {
+	command := []string{
+		"set -ex",
+		// Check NPD GPU count plugin config exists
+		"test -f /etc/node-problem-detector.d/custom-plugin-monitor/gpu_checks/custom-plugin-gpu-count.json",
+		// Verify node-problem-detector service is running
+		"systemctl is-active node-problem-detector",
```

Comment:
we don't install NPD as a part of the node bootstrapping custom script extension / customdata payload generated by AgentBaker - in production it's installed by AKSNodeExtension (aks-vm-extension).

with that in mind, how will NPD be available on AgentBaker E2E nodes? cc @ganeshkumarashok if I'm missing something

Reply:
We need to see how aks-vm-extensions can be run with AB E2E. There's two options for the extension version that runs:
1) Use extension version in the relevant PR akin to what the current NPD manual testing has (but making it automated)
2) Use latest version of the extension that is available in production - testing cycle will be delayed (by a week or so), so the dev loop is bad. But that can be a start.

Reply:
Exact code changes need to be investigated - but since AB E2E is internal and can be configured to access the resources, it should be possible IMO.